cmds/dhclient/dhclient.go

```diff
@@ -205,26 +207,36 @@ func dhclient4(iface netlink.Link, numRenewals int, timeout time.Duration, retry
 // dhcp6 support in go is hard to find. This function represents our best current
 // guess based on reading and testing.
 func dhclient6(iface netlink.Link, numRenewals int, timeout time.Duration, retry int) error {
-	conn, err := dhcp6.NewPacketSock(iface.Attrs().Index)
+	client, err := dhcp6client.New(iface, timeout, retry)
 	if err != nil {
-		return fmt.Errorf("client connection generation: %v", err)
+		return err
 	}
-	client := dhcp6.New(iface.Attrs().HardwareAddr, conn, timeout, retry)
 
 	for i := 0; numRenewals < 0 || i < numRenewals+1; i++ {
 		debug("Start getting or renewing DHCPv6 lease")
-		iaAddrs, packet, err := client.Solicit()
+		packet, err := client.Solicit()
 		if err != nil {
 			return fmt.Errorf("error: %v", err)
 		}
 		debug("Packet: %+v\n\n", packet)
+		// get IA_NA
+		iana, err := dhcp6opts.GetIANA(packet.Options)
+		if err != nil {
+			return err
+		}
+		// get IAAddrs from IA_NA
+		iaAddrs, err := dhcp6opts.GetIAAddr(iana[0].Options)
+		if err != nil {
+			return err
+		}
 		debug("IAAddrs: %v\n", iaAddrs)
 
 		if *test == false {
+			mask := net.IPMask(net.ParseIP("ffff:ffff:ffff:ffff::"))
 			dst := &netlink.Addr{
-				IPNet:       &net.IPNet{IP: iaAddrs[0].IP},
-				PreferedLft: int(iaAddrs[0].PreferredLifetime.Seconds()),
-				ValidLft:    int(iaAddrs[0].ValidLifetime.Seconds()),
+				IPNet:       &net.IPNet{IP: iaAddrs[0].IP, Mask: mask},
+				PreferedLft: int(iana[0].T1.Seconds()),
```

Comment:
I think this should still be `iaAddrs[0].PreferredLifetime`, and below.

Reply:
right